src/meltano/core/project_files.py

```diff
@@ -214,11 +295,39 @@ def _split_config_dict(self, config: dict):
                 self._add_environments(file_dicts, value)
             else:
                 file = str(self._meltano_file_path)
-                file_dict = file_dicts.setdefault(file, {})
+                file_dict = file_dicts.setdefault(file, CommentedMap())
                 file_dict[key] = value
+
+        config.copy_attributes(file_dicts[str(self._meltano_file_path)])
+        self._copy_yaml_attributes(file_dicts)
         return file_dicts
 
-    @staticmethod
-    def _write_file(file_path, contents):
+    def _copy_yaml_attributes(  # noqa: WPS210
+        self,
+        file_dicts: dict[str, CommentedMap],
+    ):
+        for file, file_dict in file_dicts.items():
```

Comment:
There is a bit going on in here, as this seems to do most of the comment preservation work. Do you mind adding a docstring with a brief description for future skimming?